### PR TITLE
Force migration support

### PR DESCRIPTION
**Pull request by mauroservienti**

### CODE REVIEW

**Comment by lailabougria** [comment]
Not sure if it makes sense from the user perspective having to run the tool with --forceMigration when they've successfully completed a previous run (could be a prime example of idempotent runs here :p). They might be rerunning to check all the timeouts were migrated or they might be running with a shorter cutoff time to maybe migrate even more timeouts than the previous run. From the user perspective I would expect hat if I've started a migration and stopped along the way due to failure or killing the process, that I can 
- continue from the previous state 
- restore whatever was done to the pre-migration state
If the last migration complete successfully, I'd expect the system to react as if I've never before ran a migration and just reanalyse whatever is in the storage but maybe add some info about the previous run just to point that out.

Might be interesting to add some feedback at the end of a successful migration as to what timeouts were migrated.

**Comment by andreasohlund** [comment]
>Not sure if it makes sense from the user perspective having to run the tool with --forceMigration when they've successfully completed a previous run 

👍 I think a clean run should not require this

>Might be interesting to add some feedback at the end of a successful migration as to what timeouts were migrated.

yea, number of migrates timeouts should IMO be printed

File: src/TimeoutMigrationTool/MigrationRunner.cs [hunk 1]
```diff
@@ -17,42 +17,65 @@ public MigrationRunner(ITimeoutStorage timeoutStorage, ICreateTransportTimeouts
...
+            if (toolState != null)
             {
-                toolState = new ToolState(runParameters);
-                await timeoutStorage.StoreToolState(toolState).ConfigureAwait(false);
+                await Console.Out.WriteLineAsync("Found existing tool state.").ConfigureAwait(false);
+                await Console.Out.WriteLineAsync("Checking migration status.").ConfigureAwait(false);
+
+                if (toolState.Status == MigrationStatus.Completed)
+                {
+                    await Console.Out.WriteLineAsync("Migration already completed. To rerun a migration on this timeout storage run the tool using --forceMigration to reset the migration state.").ConfigureAwait(false);
```
Comment (andreasohlund) [inline src/TimeoutMigrationTool/MigrationRunner.cs:37]: I think we should just allow this to continue since the previous run was completed

**Comment by andreasohlund** [comment]
@mauroservienti I've rebased this and applied the changes @lailabougria mentions in https://github.com/Particular/TimeoutMigrationTool/pull/20#issuecomment-632497370

**Pull request merged**